addon/components/nypr-accounts/membership-card/index.js

```diff
@@ -0,0 +1,46 @@
+import Ember from 'ember';
+import computed from 'ember-computed';
+import layout
+  from '../../../templates/components/nypr-accounts/membership-card';
+
+export default Ember.Component.extend({
+  layout,
+  orderDateSorting: ['orderDate:desc'],
+  sortedPledges: computed.sort('pledges', 'orderDateSorting'),
+  sortedSustainingPledges: computed.filterBy('sortedPledges', 'orderType', 'sustainer'),
+  sortedOneTimePledges: computed.filterBy('sortedPledges', 'orderType', 'onetime'),
+  mostRecentPledge: computed.reads('sortedPledges.firstObject'),
+  activePledges: computed.filterBy('sortedPledges', 'isActiveMember', true),
+  activeSustainingPledges: computed.filterBy(
+    'activePledges',
+    'orderType',
+    'sustainer'
+  ),
+  activeOneTimePledges: computed.filterBy(
+    'activePledges',
+    'orderType',
+    'onetime'
+  ),
+  isLapsedOneTimeMember: computed('pledges', 'activePledges', function() {
+    return (
+      this.get('sortedPledges.length') > 0 &&
+      this.get('activePledges').length === 0
+    );
+  }),
+  showPaymentHistory: false,
+  mostRecentSustainingPledgesPerOrderCode: computed(
+    'activeSustainingPledges',
+    function() {
+      let orderCodes = [];
+      let latestPledgePerOrderCode = [];
+      this.get('activeSustainingPledges').forEach(pledge => {
```

Comment:
Seems to be an empty array in the event of no active sustaining pledges, and it works fine in practice when there are no active sustaining pledges (it shows the 'you last donated on X' screen).